scripts/gen_lsp.lua

```diff
@@ -265,26 +283,39 @@ end
 
 local opt = {
   output_file = 'runtime/lua/vim/lsp/_meta/protocol.lua',
-  version = nil,
-  methods = nil,
+  version = DEFAULT_LSP_VERSION,
+  methods = false,
 }
 
-for i = 1, #_G.arg do
+local command = nil
+local i = 1
+while i <= #_G.arg do
   if _G.arg[i] == '--out' then
-    opt.output_file = _G.arg[i + 1]
+    opt.output_file = assert(_G.arg[i + 1], '--out <outfile> needed')
+    i = i + 1
   elseif _G.arg[i] == '--version' then
-    opt.version = _G.arg[i + 1]
+    opt.version = assert(_G.arg[i + 1], '--version <version> needed')
+    i = i + 1
   elseif _G.arg[i] == '--methods' then
     opt.methods = true
-  elseif vim.startswith(_G.arg[i], '--') then
-    opt.output_file = _G.arg[i]:sub(3)
+  elseif vim.startswith(_G.arg[i], '-') then
+    error('Unrecognized args: ' .. _G.arg[i])
+  else
+    if command then
+      error('More than one command was given: ' .. _G.arg[i])
+    else
+      command = _G.arg[i]
+    end
```

Comment:
is the nesting needed?
```suggestion
  elseif command then
    error('More than one command was given: ' .. _G.arg[i])
  else
    command = _G.arg[i]
```

Reply:
I prefer having one nested if-else here because L304-L308 altogether corresponds to handling of the "positional argument" `command`, but I don't have a strong opinion.